### DISCUSSION TITLE
[Feature] Expose additional EXIF fields for searching

### DISCUSSION THREAD

**Discussion by cycneuramus** [immich-app/immich#1752] [Feature Request]
### The feature

Hi everyone and thank you very much for this awesome piece of free software!

Say I am using a photo management app such as DigiKam to recognize faces or pick tags and write the corresponding EXIF data (`Subject`, `Region Name`, `Tags List`, `Keywords` etc.) to my photos before sending them off to Immich. Wouldn't it be nice if Immich could search through these EXIF fields in addition to the ones already supported, such as the camera/lens model?

This would mean I could use Immich to find persons effortlessly—or photos matching any other tag, really—without having to rely on server-side implementations of facial/object recognition. Yes, this presupposes a well-organized photo library; but I know there are quite a few people besides me who does this, and given a feature such as [Gallery Watcher](https://github.com/immich-app/immich/discussions/1006), I think this would be a no-brainer.

I'm not familiar with TypeScript/Dart, nor do I have much time to spare at the moment, otherwise I'd love to have a crack at this. As far as I can tell, though, this should not be too difficult to implement. What do you all think?

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by alextran1502** [comment]
Hello, are these properties preset, meaning the key are uniform or is it customized by the user in DigiKam?

**Reply by cycneuramus** [reply]
They are supposed to be uniform as far as I know; DigiKam was just an example here. As an aside, this feature (looking up photos by these kinds of EXIF tags) is present in PiGallery2, also written in TS. [This](https://github.com/bpatrik/pigallery2/blob/d11719cc3981b5481abe4333f63ad53fa74edd05/src/backend/model/threading/MetadataLoader.ts#L262) seems to be a relevant part of the source code as far as extracting the metadata goes.

**Comment by cycneuramus** [comment]
For reference, [here](https://github.com/immich-app/immich/pull/1958) is a pull request tracking this feature.
